### PR TITLE
Update documentation folder, dependencies and deployment action

### PR DESCRIPTION
**Pull request by ADubhlaoich**
### Proposed changes

This commit makes a number of changes related to documentation:

- It changes the folder from /docs to /site, matching other NGINX software, which retain /docs for development documentation
- It updates go.mod with the new theme file
- It updates the Docs section of the .gitignore file
- It updates the deployment action to match the folder change, and adds the ability to change the theme version during deployment

Notably, the latest theme update has a new shortcode which allows for code files to be embedded from GitHub links, creating the possibility of single sources of truth in documentation examples.

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [ ] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that all unit tests pass after adding my changes
- [ ] I have updated necessary documentation
- [ ] I have rebased my branch onto main
- [ ] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

**Review by j1m-ryan — COMMENTED** [review]
Theres two CI changes needed for this. One is here on line 63 and 64, ^docs should be ^site
https://github.com/nginxinc/kubernetes-ingress/blob/94b5b8a84fbee8907c536bf13ab119f88e46bc25/.github/workflows/ci.yml#L60
The other is here https://github.com/nginxinc/kubernetes-ingress/blob/94b5b8a84fbee8907c536bf13ab119f88e46bc25/.github/workflows/ci.yml#L207
cd into site instead

**Review by pdabelf5 — COMMENTED** [review]
Looks like there are a few more docs directory references

```
33 results - 19 files

.dockerignore:
  1: docs*

.fossa.yml:
  7:     - docs

.gitignore:
  52: # Ignore local docs dependencies
  55: docs/public
  56: docs/themes/f5-hugo
  58: docs/.netlify

.markdownlint-cli2.yaml:
  13:   - "docs/**" # Ignore docs folder for now

.pre-commit-config.yaml:
   3: exclude: (^docs/_vendor/|.*pb2.*)
  11:         exclude: docs/layouts/shortcodes/nic-.*.html

CODEOWNERS:
  2: /docs/ @nginxinc/kic @nginxinc/nginx-docs

CONTRIBUTING.md:
  31: Follow our [Installation Guide](https://github.com/nginxinc/kubernetes-ingress/blob/main/docs/content/installation) to
  34: Read the [documentation](https://github.com/nginxinc/kubernetes-ingress/tree/main/docs) and

README.md:
  144: | Edge/Nightly | For testing and experimenting | Use the edge or nightly images from [DockerHub](https://hub.docker.com/r/nginx/nginx-ingress/), [GitHub Container](https://github.com/nginxinc/kubernetes-ingress/pkgs/container/kubernetes-ingress), [Amazon ECR Public Gallery](https://gallery.ecr.aws/nginx/nginx-ingress) or [Quay.io](https://quay.io/repository/nginx/nginx-ingress) or [build your own image](https://docs.nginx.com/nginx-ingress-controller/installation/building-nginx-ingress-controller/). | [Build your own image](https://docs.nginx.com/nginx-ingress-controller/installation/building-nginx-ingress-controller/). | [Manifests](https://github.com/nginxinc/kubernetes-ingress/tree/main/deployments). [Helm chart](https://github.com/nginxinc/kubernetes-ingress/tree/main/charts/nginx-ingress). | [Documentation](https://github.com/nginxinc/kubernetes-ingress/tree/main/docs/content). [Examples](https://github.com/nginxinc/kubernetes-ingress/tree/main/examples). |

.github/dependabot.yml:
  59:     directory: /docs

.github/labeler.yml:
  35:   - head-branch: ['^docs/', '^doc/']

.github/scripts/release-notes-update.sh:
  9: DOCS_TO_UPDATE_FOLDER=${ROOTDIR}/docs/content

.github/scripts/release-version-update.sh:
   12: DOCS_TO_UPDATE_FOLDER=${ROOTDIR}/docs/content
  124: echo -n "${new_ic_version}" > ./docs/layouts/shortcodes/nic-version.html
  125: echo -n "${new_helm_chart_version}" > ./docs/layouts/shortcodes/nic-helm-version.html
  126: echo -n "${new_operator_version}" > ./docs/layouts/shortcodes/nic-operator-version.html

.github/scripts/variables.sh:
  21:   find . -type f \( -name "*.go" -o -name go.mod -o -name go.sum -o -name "*.tmpl" -o -name "version.txt" \) -not -path "./docs*"  -exec md5sum {} + | LC_ALL=C sort  | md5sum | awk '{ print $1 }'

.github/workflows/ci.yml:
   63:           files=$(git diff --name-only HEAD^ | egrep -v "^docs/" | egrep -v "^examples/" | egrep -v "^README.md")
   64:           docs_files=$(git diff --name-only HEAD^ | grep "^docs/")
  209:         run: cd docs && make docs-ci

.github/workflows/codeql-analysis.yml:
  38:           files=$(git diff --name-only HEAD^ | egrep -v "^docs/" | egrep -v "^examples/" | egrep -v "^README.md")

.github/workflows/dependabot-hugo.yml:
   6:       - "docs/go.mod"
  41:         working-directory: docs

.github/workflows/docs-build-push.yml:
  19:       - "docs/**"
  37:       docs_build_path: "./docs"

.github/workflows/fossa.yml:
  8:       - "docs/**"

.github/workflows/mend.yml:
  8:       - docs/**

```

**Comment by ADubhlaoich** [comment]
@pdabelf5 GRMA, pushed changes!

Learned a lot about the CI/CD while going through everything - I'd really like to fork a version of the cherry-pick workflow for elsewhere in the org.

File: .github/labeler.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ tests:
       - all-globs-to-all-files: ['!tests/requirements.txt', '!perf-tests/requirements.txt']
 
 documentation:
-  - head-branch: ['^docs/', '^doc/']
+  - head-branch: ['^site/']
```
Comment (pdabelf5) [inline .github/labeler.yml:35]: @ADubhlaoich I'm re-thinking this as I've taken a second look.  This actually means branches named `docs/mybranch` which I think we want to keep.  Apologies for the mistake.

File: .github/scripts/release-version-update.sh [hunk 2]
```diff
@@ -9,7 +9,7 @@ DEPLOYMENT_PATH="${ROOTDIR}/deployments"
 EXAMPLES_PATH="${ROOTDIR}/examples"
 DEBUG=${DEBUG:-"false"}
 
-DOCS_TO_UPDATE_FOLDER=${ROOTDIR}/docs/content
+DOCS_TO_UPDATE_FOLDER=${ROOTDIR}/docs/site
```
Comment (pdabelf5) [inline .github/scripts/release-version-update.sh:12]:
```suggestion
DOCS_TO_UPDATE_FOLDER=${ROOTDIR}/site/content
```

**Review by pdabelf5 — APPROVED** [review]

**Pull request merged**